ingestion/src/metadata/ingestion/source/metadata/alationsink/client.py

```diff
@@ -0,0 +1,207 @@
+            api_version="integration/v1",
+            auth_token=lambda: ("no_token", 0),
+            auth_header="Authorization",
+            allow_redirects=True,
+        )
+        self.client = REST(client_config)
+        self.generated_auth_token = None
+        self.expiry = None
+        super().__init__()
+
+    @classmethod
+    def create(cls, config: AlationSinkConnection):
+        return cls(config)
+
+    def auth_token(self) -> None:
+        """
+        Generate the auth token
+        """
+        if isinstance(self.config.authType, ApiAccessTokenAuth):
+            self.generated_auth_token = (
+                self.config.authType.accessToken.get_secret_value()
+            )
+            self.expiry = 0
+        else:
+            self._get_access_token_from_basic_auth()
+
+    def get_access_token(self):
+        """
+        Return the generated access token
+        """
+        self.auth_token()
+        return self.generated_auth_token, self.expiry
+
+    def _get_access_token_from_basic_auth(self):
+        """
+        Get the access token for basic auth type
+        """
+        # Get the refresh token
+        refresh_token_data = {
+            "username": self.config.authType.username,
+            "password": self.config.authType.password.get_secret_value(),
+            "name": self.config.projectName,
+        }
+        refresh_token_response = self.client.post(
+            "/createRefreshToken/", json.dumps(refresh_token_data)
+        )
+
+        # Get the access token
+        access_token_data = {
+            "refresh_token": refresh_token_response["refresh_token"],
+            "user_id": refresh_token_response["user_id"],
+        }
+        access_token_response = self.client.post(
+            "/createAPIAccessToken/", json.dumps(access_token_data)
+        )
+
+        self.generated_auth_token = access_token_response["api_access_token"]
+        self.expiry = 0
+
+
+class AlationSinkClient:
+    """
+    Client to interact with Alation apis
+    """
+
+    def __init__(self, config: AlationSinkConnection):
+        self.config = config
+        self._auth_provider = AlationSinkAuthenticationProvider.create(config)
+        client_config: ClientConfig = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            auth_header="TOKEN",
+            api_version="integration",
+            auth_token=self._auth_provider.get_access_token,
+            auth_token_mode="",
+            verify=False,
+        )
+        self.client = REST(client_config)
+        self.pagination_limit = self.config.paginationLimit
+
+    def paginate_entity(
+        self, api_url: str, data: Optional[Dict] = None, is_key_offset: bool = False
+    ) -> Optional[List[Any]]:
+        """
+        Method to paginate the entities
+        """
+        skip_key_name = "skip"
+        if is_key_offset:
+            skip_key_name = "offset"
+        entities = []
+        total_records = 100000000
```

Comment:
this might be better a global var, easier to control? Plus is this the records we pick in one go?

Reply:
Since the alation APIs do not provide us how many total records are present, we are setting a upper limit on the total records.

The pagination logic will always and break the loop if the next API does not return any results

Reply:
ok thanks, lets have this as a constant with this comment at the top